File: Models/Monica_Evapotranspiration/src/sirius/Monica_Evapotranspiration/EvapotranspirationCompState.cs

using System;
using System.Collections.Generic;
using CRA.ModelLayer.Core;
using System.Reflection;
using CRA.ModelLayer.ParametersManagement;   

namespace SiriusQualityEvapotranspirationComp.DomainClass
{
    public class EvapotranspirationCompState : ICloneable, IDomainClass
    {
        private double _evaporated_from_surface;
        private double _surface_water_storage;
        private double _snow_depth;
        private int _developmental_stage;
        private double _crop_reference_evapotranspiration;
        private double _reference_evapotranspiration;
        private double _actual_evaporation;
        private double _actual_transpiration;
        private double _kc_factor;
        private double _percentage_soil_coverage;
        private double[] _soil_moisture;
        private double[] _permanent_wilting_point;
        private double[] _field_capacity;
        private double[] _evaporation;
        private double[] _transpiration;
        private double[] _crop_transpiration;
        private double _crop_remaining_evapotranspiration;
        private double _crop_evaporated_from_intercepted;
        private double[] _evapotranspiration;
        private double _actual_evapotranspiration;
        private double _vapor_pressure;
        private ParametersIO _parametersIO;

        public EvapotranspirationCompState()
        {
            _parametersIO = new ParametersIO(this);
        }

        public EvapotranspirationCompState(EvapotranspirationCompState toCopy, bool copyAll) // copy constructor 
        {
            if (copyAll)
            {
                        evaporated_from_surface = toCopy.evaporated_from_surface;
                        surface_water_storage = toCopy.surface_water_storage;
                        snow_depth = toCopy.snow_depth;
                        developmental_stage = toCopy.developmental_stage;
                        crop_reference_evapotranspiration = toCopy.crop_reference_evapotranspiration;
                        reference_evapotranspiration = toCopy.reference_evapotranspiration;
                        actual_evaporation = toCopy.actual_evaporation;
                        actual_transpiration = toCopy.actual_transpiration;
                        kc_factor = toCopy.kc_factor;
                        percentage_soil_coverage = toCopy.percentage_soil_coverage;
                        soil_moisture = new double[toCopy.soil_moisture.Length];
        for (int i = 0; i < toCopy.soil_moisture.Length; i++)
            { soil_moisture[i] = toCopy.soil_moisture[i]; }
    
                        permanent_wilting_point = new double[toCopy.permanent_wilting_point.Length];
        for (int i = 0; i < toCopy.permanent_wilting_point.Length; i++)
            { permanent_wilting_point[i] = toCopy.permanent_wilting_point[i]; }
    
                        field_capacity = new double[toCopy.field_capacity.Length];
        for (int i = 0; i < toCopy.field_capacity.Length; i++)
            { field_capacity[i] = toCopy.field_capacity[i]; }
    
                        evaporation = new double[toCopy.evaporation.Length];
        for (int i = 0; i < toCopy.evaporation.Length; i++)
            { evaporation[i] = toCopy.evaporation[i]; }
    
                        transpiration = new double[toCopy.transpiration.Length];
        for (int i = 0; i < toCopy.transpiration.Length; i++)
            { transpiration[i] = toCopy.transpiration[i]; }
    
                        crop_transpiration = new double[toCopy.crop_transpiration.Length];
        for (int i = 0; i < toCopy.crop_transpiration.Length; i++)
            { crop_transpiration[i] = toCopy.crop_transpiration[i]; }
    
                        crop_remaining_evapotranspiration = toCopy.crop_remaining_evapotranspiration;
                        crop_evaporated_from_intercepted = toCopy.crop_evaporated_from_intercepted;
                        evapotranspiration = new double[toCopy.evapotranspiration.Length];
        for (int i = 0; i < toCopy.evapotranspiration.Length; i++)
            { evapotranspiration[i] = toCopy.evapotranspiration[i]; }
    
                        actual_evapotranspiration = toCopy.actual_evapotranspiration;
                        vapor_pressure = toCopy.vapor_pressure;
                    }
                }

                public double evaporated_from_surface
    {
        get { return this._evaporated_from_surface; }
        set { this._evaporated_from_surface= value; } 
    }
                public double surface_water_storage
    {
        get { return this._surface_water_storage; }
        set { this._surface_water_storage= value; } 
    }
                public double snow_depth
    {
        get { return this._snow_depth; }
        set { this._snow_depth= value; } 
    }
                public int developmental_stage
    {
        get { return this._developmental_stage; }
        set { this._developmental_stage= value; } 
    }
                public double crop_reference_evapotranspiration
    {
        get { return this._crop_reference_evapotranspiration; }
        set { this._crop_reference_evapotranspiration= value; } 
    }
                public double reference_evapotranspiration
    {
        get { return this._reference_evapotranspiration; }
        set { this._reference_evapotranspiration= value; } 
    }
                public double actual_evaporation
    {
        get { return this._actual_evaporation; }
        set { this._actual_evaporation= value; } 
    }
                public double actual_transpiration
    {
        get { return this._actual_transpiration; }
        set { this._actual_transpiration= value; } 
    }
                public double kc_factor
    {
        get { return this._kc_factor; }
        set { this._kc_factor= value; } 
    }
                public double percentage_soil_coverage
    {
        get { return this._percentage_soil_coverage; }
        set { this._percentage_soil_coverage= value; } 
    }
                public double[] soil_moisture
    {
        get { return this._soil_moisture; }
        set { this._soil_moisture= value; } 
    }
                public double[] permanent_wilting_point
    {
        get { return this._permanent_wilting_point; }
        set { this._permanent_wilting_point= value; } 
    }
                public double[] field_capacity
    {
        get { return this._field_capacity; }
        set { this._field_capacity= value; } 
    }
                public double[] evaporation
    {
        get { return this._evaporation; }
        set { this._evaporation= value; } 
    }
                public double[] transpiration
    {
        get { return this._transpiration; }
        set { this._transpiration= value; } 
    }
                public double[] crop_transpiration
    {
        get { return this._crop_transpiration; }
        set { this._crop_transpiration= value; } 
    }
                public double crop_remaining_evapotranspiration
    {
        get { return this._crop_remaining_evapotranspiration; }
        set { this._crop_remaining_evapotranspiration= value; } 
    }
                public double crop_evaporated_from_intercepted
    {
        get { return this._crop_evaporated_from_intercepted; }
        set { this._crop_evaporated_from_intercepted= value; } 
    }
                public double[] evapotranspiration
    {
        get { return this._evapotranspiration; }
        set { this._evapotranspiration= value; } 
    }
                public double actual_evapotranspiration
    {
        get { return this._actual_evapotranspiration; }
        set { this._actual_evapotranspiration= value; } 
    }
                public double vapor_pressure
    {
        get { return this._vapor_pressure; }
        set { this._vapor_pressure= value; } 
    }

                public string Description
                {
                    get { return "EvapotranspirationCompState of the component";}
                }

                public string URL
                {
                    get { return "http://" ;}
                }

                public virtual IDictionary<string, PropertyInfo> PropertiesDescription
                {
                    get { return _parametersIO.GetCachedProperties(typeof(IDomainClass));}
                }

                public virtual Boolean ClearValues()
                {
                     _evaporated_from_surface = default(double);
                     _surface_water_storage = default(double);
                     _snow_depth = default(double);
                     _developmental_stage = default(int);
                     _crop_reference_evapotranspiration = default(double);
                     _reference_evapotranspiration = default(double);
                     _actual_evaporation = default(double);
                     _actual_transpiration = default(double);
                     _kc_factor = default(double);
                     _percentage_soil_coverage = default(double);
                     _soil_moisture = new double[no_of_soil_layers];
                     _permanent_wilting_point = new double[no_of_soil_layers];
                     _field_capacity = new double[no_of_soil_layers];
                     _evaporation = new double[no_of_soil_layers];
                     _transpiration = new double[no_of_soil_layers];
                     _crop_transpiration = new double[no_of_soil_layers];
                     _crop_remaining_evapotranspiration = default(double);
                     _crop_evaporated_from_intercepted = default(double);
                     _evapotranspiration = new double[no_of_soil_layers];
                     _actual_evapotranspiration = default(double);
                     _vapor_pressure = default(double);
                    return true;
                }

                public virtual Object Clone()
                {
                    IDomainClass myclass = (IDomainClass) this.MemberwiseClone();
                    _parametersIO.PopulateClonedCopy(myclass);
                    return myclass;
                }
            }
        }

File: Models/Monica_Evapotranspiration/src/java/Monica_Evapotranspiration/EvapotranspirationCompState.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class EvapotranspirationCompState
{
    private double evaporated_from_surface;
    private double surface_water_storage;
    private double snow_depth;
    private Integer developmental_stage;
    private double crop_reference_evapotranspiration;
    private double reference_evapotranspiration;
    private double actual_evaporation;
    private double actual_transpiration;
    private double kc_factor;
    private double percentage_soil_coverage;
    private Double [] soil_moisture;
    private Double [] permanent_wilting_point;
    private Double [] field_capacity;
    private Double [] evaporation;
    private Double [] transpiration;
    private Double [] crop_transpiration;
    private double crop_remaining_evapotranspiration;
    private double crop_evaporated_from_intercepted;
    private Double [] evapotranspiration;
    private double actual_evapotranspiration;
    private double vapor_pressure;
    
    public EvapotranspirationCompState() { }
    
    public EvapotranspirationCompState(EvapotranspirationCompState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.evaporated_from_surface = toCopy.getevaporated_from_surface();
            this.surface_water_storage = toCopy.getsurface_water_storage();
            this.snow_depth = toCopy.getsnow_depth();
            this.developmental_stage = toCopy.getdevelopmental_stage();
            this.crop_reference_evapotranspiration = toCopy.getcrop_reference_evapotranspiration();
            this.reference_evapotranspiration = toCopy.getreference_evapotranspiration();
            this.actual_evaporation = toCopy.getactual_evaporation();
            this.actual_transpiration = toCopy.getactual_transpiration();
            this.kc_factor = toCopy.getkc_factor();
            this.percentage_soil_coverage = toCopy.getpercentage_soil_coverage();
            soil_moisture = new Double[toCopy.getsoil_moisture().length];
        for (int i = 0; i < toCopy.getsoil_moisture().length; i++)
        {
            soil_moisture[i] = toCopy.getsoil_moisture()[i];
        }
            permanent_wilting_point = new Double[toCopy.getpermanent_wilting_point().length];
        for (int i = 0; i < toCopy.getpermanent_wilting_point().length; i++)
        {
            permanent_wilting_point[i] = toCopy.getpermanent_wilting_point()[i];
        }
            field_capacity = new Double[toCopy.getfield_capacity().length];
        for (int i = 0; i < toCopy.getfield_capacity().length; i++)
        {
            field_capacity[i] = toCopy.getfield_capacity()[i];
        }
            evaporation = new Double[toCopy.getevaporation().length];
        for (int i = 0; i < toCopy.getevaporation().length; i++)
        {
            evaporation[i] = toCopy.getevaporation()[i];
        }
            transpiration = new Double[toCopy.gettranspiration().length];
        for (int i = 0; i < toCopy.gettranspiration().length; i++)
        {
            transpiration[i] = toCopy.gettranspiration()[i];
        }
            crop_transpiration = new Double[toCopy.getcrop_transpiration().length];
        for (int i = 0; i < toCopy.getcrop_transpiration().length; i++)
        {
            crop_transpiration[i] = toCopy.getcrop_transpiration()[i];
        }
            this.crop_remaining_evapotranspiration = toCopy.getcrop_remaining_evapotranspiration();
            this.crop_evaporated_from_intercepted = toCopy.getcrop_evaporated_from_intercepted();
            evapotranspiration = new Double[toCopy.getevapotranspiration().length];
        for (int i = 0; i < toCopy.getevapotranspiration().length; i++)
        {
            evapotranspiration[i] = toCopy.getevapotranspiration()[i];
        }
            this.actual_evapotranspiration = toCopy.getactual_evapotranspiration();
            this.vapor_pressure = toCopy.getvapor_pressure();
        }
    }
    public double getevaporated_from_surface()
    { return evaporated_from_surface; }

    public void setevaporated_from_surface(double _evaporated_from_surface)
    { this.evaporated_from_surface= _evaporated_from_surface; } 
    
    public double getsurface_water_storage()
    { return surface_water_storage; }

    public void setsurface_water_storage(double _surface_water_storage)
    { this.surface_water_storage= _surface_water_storage; } 
    
    public double getsnow_depth()
    { return snow_depth; }

    public void setsnow_depth(double _snow_depth)
    { this.snow_depth= _snow_depth; } 
    
    public Integer getdevelopmental_stage()
    { return developmental_stage; }

    public void setdevelopmental_stage(Integer _developmental_stage)
    { this.developmental_stage= _developmental_stage; } 
    
    public double getcrop_reference_evapotranspiration()
    { return crop_reference_evapotranspiration; }

    public void setcrop_reference_evapotranspiration(double _crop_reference_evapotranspiration)
    { this.crop_reference_evapotranspiration= _crop_reference_evapotranspiration; } 
    
    public double getreference_evapotranspiration()
    { return reference_evapotranspiration; }

    public void setreference_evapotranspiration(double _reference_evapotranspiration)
    { this.reference_evapotranspiration= _reference_evapotranspiration; } 
    
    public double getactual_evaporation()
    { return actual_evaporation; }

    public void setactual_evaporation(double _actual_evaporation)
    { this.actual_evaporation= _actual_evaporation; } 
    
    public double getactual_transpiration()
    { return actual_transpiration; }

    public void setactual_transpiration(double _actual_transpiration)
    { this.actual_transpiration= _actual_transpiration; } 
    
    public double getkc_factor()
    { return kc_factor; }

    public void setkc_factor(double _kc_factor)
    { this.kc_factor= _kc_factor; } 
    
    public double getpercentage_soil_coverage()
    { return percentage_soil_coverage; }

    public void setpercentage_soil_coverage(double _percentage_soil_coverage)
    { this.percentage_soil_coverage= _percentage_soil_coverage; } 
    
    public Double [] getsoil_moisture()
    { return soil_moisture; }

    public void setsoil_moisture(Double [] _soil_moisture)
    { this.soil_moisture= _soil_moisture; } 
    
    public Double [] getpermanent_wilting_point()
    { return permanent_wilting_point; }

    public void setpermanent_wilting_point(Double [] _permanent_wilting_point)
    { this.permanent_wilting_point= _permanent_wilting_point; } 
    
    public Double [] getfield_capacity()
    { return field_capacity; }

    public void setfield_capacity(Double [] _field_capacity)
    { this.field_capacity= _field_capacity; } 
    
    public Double [] getevaporation()
    { return evaporation; }

    public void setevaporation(Double [] _evaporation)
    { this.evaporation= _evaporation; } 
    
    public Double [] gettranspiration()
    { return transpiration; }

    public void settranspiration(Double [] _transpiration)
    { this.transpiration= _transpiration; } 
    
    public Double [] getcrop_transpiration()
    { return crop_transpiration; }

    public void setcrop_transpiration(Double [] _crop_transpiration)
    { this.crop_transpiration= _crop_transpiration; } 
    
    public double getcrop_remaining_evapotranspiration()
    { return crop_remaining_evapotranspiration; }

    public void setcrop_remaining_evapotranspiration(double _crop_remaining_evapotranspiration)
    { this.crop_remaining_evapotranspiration= _crop_remaining_evapotranspiration; } 
    
    public double getcrop_evaporated_from_intercepted()
    { return crop_evaporated_from_intercepted; }

    public void setcrop_evaporated_from_intercepted(double _crop_evaporated_from_intercepted)
    { this.crop_evaporated_from_intercepted= _crop_evaporated_from_intercepted; } 
    
    public Double [] getevapotranspiration()
    { return evapotranspiration; }

    public void setevapotranspiration(Double [] _evapotranspiration)
    { this.evapotranspiration= _evapotranspiration; } 
    
    public double getactual_evapotranspiration()
    { return actual_evapotranspiration; }

    public void setactual_evapotranspiration(double _actual_evapotranspiration)
    { this.actual_evapotranspiration= _actual_evapotranspiration; } 
    
    public double getvapor_pressure()
    { return vapor_pressure; }

    public void setvapor_pressure(double _vapor_pressure)
    { this.vapor_pressure= _vapor_pressure; } 
    
}